### PR TITLE
[PR ENHANCEMENT] Prevent repeated API calls on explore page's search bar by adding debounce

### LINKED ISSUE #322: [ENHANCEMENT] Search should be triggered when user stops typing
**Description:** On the Explore page, the `handleSearch` function is triggered every time the user types a character in the search box. This results in multiple unnecessary API calls, which may lead to performance issues and a poor user experience. 

The search should only be triggered when the user has stopped typing for a brief period. This ensures efficient API usage and better performance.

Related to #39 
PR resolving issue

### PR DESCRIPTION
**Pull request by bt919**
**Description:**
This PR adds a 500ms debounce to the explore page's search bar so that less API calls are made to the backend, and less loading states are shown to users as well.

**Summary of changes:**
  - Changed the onChange handler for the search bar to add debounce
  - Changed the search bar's onKeyDown and button's onClick handlers to inline (so those don't get debounced as well)
  - Moved the loading spinner further down into the component so the nav and other parts of the page still stays visible/interactive to the users

PR close #322

### CODE REVIEW

File: client/src/pages/ExplorePage.jsx [hunk 1]
```diff
@@ -257,13 +256,19 @@ const ExplorePage = () => {
...
 
-  const handleSearch = debounce((query) => {
-    fetchData(query)
-  }, 500)
+  const debouncedSearch = (query) => {
+    clearTimeout(timeoutRef.current)
+
+    const timeoutId = setTimeout(() => {
+      fetchData(query)
+    }, 200)
```
Comment (MyWaterBottle) [inline client/src/pages/ExplorePage.jsx]: I would make this value higher as 200ms is very fast. Otherwise, great addition!

**Review by MyWaterBottle — CHANGES_REQUESTED** [review]
Good addition. Would change the debounce to a higher value to accomodate most typing speed.

File: client/src/pages/ExplorePage.jsx (hunk 1 above)
Comment (bt919) [reply client/src/pages/ExplorePage.jsx]: Thanks for the feedback! should be good now

**Review by MyWaterBottle — APPROVED** [review]
Looks good! Can merge.

**Pull request merged**